packages/common/src/logger.ts

```diff
@@ -13,3 +13,44 @@ export interface Logger {
   warn(message: string, meta?: LogMetadata): any;
   error(message: string, meta?: LogMetadata): any;
 }
+
+/**
+ * Possible values of the `logSource` meta attributes on log messages. This
+ * attribute indicates which subsystem emitted the log message; this may for
+ * example be used to implement fine-grained filtering of log messages.
+ *
+ * Note that there is no guarantee that this list will remain stable in the
+ * future; values may be added or removed, and messages that are currently
+ * emitted with some `logSource` value may use a different value in the future.
+ */
+export enum LogSource {
+  /**
```

Comment:
Do you also need a "client" source value?

Reply:
No, the client side API don't use a logger.

Reply:
Not for this PR, but do we need one? How do we log client side errors?

Reply:
> Not for this PR, but do we need one? How do we log client side errors?

I don't think there's a real need to unify client side logging at the moment. Errors on client side are simply thrown back to user code.

Now, the `@grpc/grpc-js` library may technically produce log messages, and it accepts a logger. But I haven't seen anyone reach out for these messages, except when intentionally enabling gRPC's verbose/debug trace messages to diagnose a specific issue, which is not a use case we should be bothering.

Also, if we were at some point to start capturing client side logging, that couldn't go through the `Runtime` class, as we want the `@temporalio/client` package to be usable in some execution contexts where the `@temporalio/worker` package can't run. Sure, we can work this out, i.e. extract unified logger to the `common` package, but is it worth it?

If anyone really want to capture grpc's log messages, they can simply themselves calls that library's `setLogger` themselves.